api/client/rm.go

```diff
@@ -30,6 +30,7 @@ func (cli *DockerCli) CmdRm(args ...string) error {
 
 	var encounteredError error
 	for _, name := range cmd.Args() {
+		name = stripPrefix(name)
```

Comment:
Yep, I agree and I can change. Only reason for the function is that all api calls using container name are going to use it. Just paranoia to see if another preprocessing may happen in the future. Let me know if you want to to change it in place.